### PR TITLE
Adding implicit conversion from ArraySegment and TryCopyTo APIs.

### PR DESCRIPTION
**Pull request by ahsonkhan**
**Note:** This PR will only pass CI after https://github.com/dotnet/corefxlab/pull/1602 is merged.

cc @KrzysztofCwalina, @shiftylogic

### CODE REVIEW

**Comment by ahsonkhan** [comment]
@dotnet-bot test this please

**Comment by ahsonkhan** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 Debug Build and Test

**Comment by KrzysztofCwalina** [comment]
Looks good

File: tests/System.Buffers.Primitives.Tests/CopyTo.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+{
+    public class CopyTo
+    {
+        [Fact]
+        public static void CopyToSpan()
+        {
+            int[] src = { 1, 2, 3 };
+            int[] dst = { 99, 100, 101 };
+
+            var srcBuffer = new Buffer<int>(src);
```
Comment (shiftylogic) [inline tests/System.Buffers.Primitives.Tests/CopyTo.cs]: Since you added the implicit cast, you should try to exercise it here and get extra coverage.

Comment (ahsonkhan) [inline tests/System.Buffers.Primitives.Tests/CopyTo.cs]: There is already coverage for implicit cast in the equality tests in https://github.com/dotnet/corefxlab/pull/1607

I think a better place to make sure we have coverage for the implicit cast is to add some tests to CtorArray.cs and CtorArrayReadOnlyBuffer.cs. Thoughts?

**Pull request merged**